e2e-tests/tfm/cypress/e2e/journeys/deals/deals-pagination.spec.js

```diff
@@ -0,0 +1,130 @@
+import relative from '../../relativeURL';
+import pages from '../../pages';
+import { T1_USER_1 } from '../../../../../e2e-fixtures';
+
+context('User can navigate through a paginated table of deals using the pagination navigation', () => {
+  const numberOfDeals = 62;
+
+  before(() => {
+    cy.deleteAllTfmDealsFromDb();
+    cy.insertManyTfmDealsIntoDb(numberOfDeals);
+  });
+
+  beforeEach(() => {
+    cy.login(T1_USER_1);
+
+    cy.url().should('eq', relative('/deals/0'));
+    pages.dealsPage.dealsTableRows().eq(0).contains('10000001');
+    pages.dealsPage.dealsTableRows().eq(19).contains('10000020');
+    pages.dealsPage.dealsTableRows().should('have.length', 20);
+  });
+
+  after(() => {
+    cy.deleteAllTfmDealsFromDb();
+  });
+
+  it('should allow the user to navigate to the next page of the deals table', () => {
+    pages.dealsPage.pagination.next().click();
+
+    cy.url().should('eq', relative('/deals/1'));
+    pages.dealsPage.dealsTableRows().eq(0).contains('10000021');
+    pages.dealsPage.dealsTableRows().eq(19).contains('10000040');
+    pages.dealsPage.dealsTableRows().should('have.length', 20);
+  });
+
+  it('should allow the user to navigate to the last page of the deals table', () => {
+    pages.dealsPage.pagination.last().click();
+
+    cy.url().should('eq', relative('/deals/3'));
+    pages.dealsPage.dealsTableRows().eq(0).contains('10000061');
+    pages.dealsPage.dealsTableRows().eq(1).contains('10000062');
+    pages.dealsPage.dealsTableRows().should('have.length', 2);
+  });
+
+  it('should allow the user to navigate to the previous page of the deals table', () => {
+    cy.visit('/deals/3');
+    pages.dealsPage.pagination.previous().click();
+
+    cy.url().should('eq', relative('/deals/2'));
+    pages.dealsPage.dealsTableRows().eq(0).contains('10000041');
+    pages.dealsPage.dealsTableRows().eq(19).contains('10000060');
+    pages.dealsPage.dealsTableRows().should('have.length', 20);
+  });
+
+  it('should allow the user to navigate to the first page of the deals table', () => {
+    cy.visit('/deals/3');
+    pages.dealsPage.pagination.first().click();
+
+    cy.url().should('eq', relative('/deals/0'));
+    pages.dealsPage.dealsTableRows().eq(0).contains('10000001');
+    pages.dealsPage.dealsTableRows().eq(19).contains('10000020');
+    pages.dealsPage.dealsTableRows().should('have.length', 20);
+  });
+
+  it('should allow the user to navigate to a specific page of the deals table', () => {
+    pages.dealsPage.pagination.page(2).click();
+
+    cy.url().should('eq', relative('/deals/2'));
+    pages.dealsPage.dealsTableRows().eq(0).contains('10000041');
+    pages.dealsPage.dealsTableRows().eq(19).contains('10000060');
+    pages.dealsPage.dealsTableRows().should('have.length', 20);
+  });
```

Comment:
Nice work, we could create some cypress commands here to be DRY e.g

```js
cy.checkDealsTableRow({ index: 19, dealNumber: 10000060  });
cy.checkDealsTableTotalRows(20);
```

Also I think we should update the row assertions to check the deal number column specifically instead of `.contains('10000001')`, because if the deal number is present in the row, but not in the deal number column, this would still pass

Reply:
Created cypress commands and updated row assertion to check deal ID column specifically